binscripts/rvm-installer

```diff
@@ -278,8 +282,9 @@ fetch_versions()
       _url=https://${_domain}/api/v3/repos/${_account}/${_repo}/tags
       ;;
   esac
+  notice "Fetching ${_url}"
```

Comment:
Why just not `log`?

Reply:
Good question there. I discovered actually necessary to avoid emitting to the STDOUT stream in this case, because `fetch_version` captures the STDOUT output of `fetch_versions`, and assigns it to `_version`.

Reply:
Maybe use debug then? @mpapis?

Reply:
With debug, regular users / observers in Travis will not see the message.  My intent was to make sure that they can see what is being fetched, as it avoids to obscure 403 Forbidden and 410 Gone errors like in the [failed build](https://travis-ci.org/adhearsion/adhearsion/jobs/555932810) below:

<img width="685" alt="failed travis build" src="https://user-images.githubusercontent.com/208529/61052206-8a37e000-a3b0-11e9-9cc9-1af964108938.png">

I think having the extra noise may help people more quickly figure out what's going on when API calls may fail.

I otherwise would not have known through a Travis build that the 403 Forbidden was from the GitHub API and the 410 Gone was from the Bitbucket v1 API.

Reply:
I agree that producing valid error output is crucial to find problems easily. I am just not such a big fan of introducing many different logging functions, as one gets lost quickly which one should be used...

Reply:
If it helps, I can attempt to retroactively print the URL, after curl has emitted a failure.  It would look a bit more like this if I can pull it off:

> curl: (22) The requested URL returned error: 403 Forbidden
> **... Fetching https://api.github.com/repos/rvm/rvm/tags has failed.**
> curl: (22) The requested URL returned error: 410 Gone
> **... Fetching https://api.bitbucket.org/2.0/repositories/mpapis/rvm/refs/tags?sort=-name&pagelen=20 has failed.**
> Downloading https://github.com/rvm/rvm/archive/.tar.gz

Then I could make use of STDOUT / `log` to print these.  Extra noise would only be printed when a failure or recoverable failure has occurred.

Should I give that a stab?

Reply:
Ideally, it should print `Fetching https://api.github.com/repos/rvm/rvm/tags has failed (403 Forbidden)` when curl failed, so basically `Fetching {url} ({error status curl})`

Reply:
I can give that a go.   Though I'm a little nervous about printing "Fetching .." prior to the curl error output, because it requires that I first capture the curl error output off of STDERR to a variable, then print "Fetching .." along with the variable's contents.  (If any part of that process fails, then we might halt before we get to print the original curl error).

I'll give it a try as a separate commit, orphaned out of the this branch / PR.  If we like how it looks, then we can include it in this PR (that way there's fewer unnecessary travis builds).

One additional Q: Should we provide the URL upon failure only in `fetch_versions`, or for all calls to `__rvm_curl`?  I think I can handle either one.  Though if we go with the latter, we may want a more generic message like:
> Request {url} has failed ({error status curl})

For now, I'm doing to draft it up as modification only to `fetch_versions` as we discussed, but I can apply it more generically to `__rvm_curl` if we like that idea.

Reply:
head's up: I attempted this and the situation [got pretty complicated](https://github.com/sfgeorge/rvm/compare/fix-rvm-installer-fetch-version-bitbucket...fix-rvm-installer-fetch-version-bitbucket.selective-noise) - I have yet to come up with a clean solution to honoring this without using STDERR, since STDOUT is reserved for emitting the version from `fetch_version`.

Since this is a difficult topic that requires more consideration, I am going to close this PR and split it into 2 distinct purposes 1.) Fix Bitbucket (urgent due to many GitHub 403 rate-limiting errors on Travis) and 2.) Improve error detection & messaging.